client/src/app/components/input-geo/input-geo.component.ts

```diff
@@ -0,0 +1,197 @@
+import { 
+  Component, OnInit, Input, Output, EventEmitter, HostBinding,
+  Optional, Self, ViewChild, ElementRef
+} from '@angular/core';
+import { FormGroup, NgControl, FormBuilder } from '@angular/forms';
+import { MatFormFieldControl } from '@angular/material/form-field';
+import { Subject } from "rxjs";
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { HttpClient } from '@angular/common/http';
+
+import { environment } from 'src/environments/environment';
+import { config } from './config';
+import { get as getScript } from 'scriptjs';
+import { User } from 'src/app/models/user.model';
+import { AuthService } from 'src/app/services/auth.service';
+declare const google: any;
+
+@Component({
```

Comment:
General: Please add JSDoc. It doesn't take too much time but makes the code much more readable due to the docstrings breaking it up.

Reply:
It makes sense, I would add this to the "refactoring" stage (since we have a lot of code wo/docs), maybe next week?